### PR TITLE
fix: DropdownButton missing properties & docs fix for Dropdown & Button

### PR DESCRIPTION
**Pull request by anddrzejb**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
`DropdownButton` is missing `Button` standard behaviours like `Ghost`, `Danger`, etc.

### 💡 Background and solution
This is a follow-up to my PR #1588.
You will see a lot of changes here, because I was also try to follow the suggestion in #1495. 
So this PR adds:
1. All missing functionality of a regular `Button` to `DropdownButton`:
   - `Block`
   - `Danger`
   - `Ghost`
   - `Loading`
   - `Type` - was already existing but this PR adds ability to set type separately for left and right button
   - `ButtonsRender` - I decided to hide this in `Dropdown` by making it `internal` and expose it only in `DropdownButton` - since this is strictly related to `DropdownButton`, I do not think it should be visible from regular `Dropdown`.
2. Adds new demos and API docs
3. Adds new button type - `text` to `Button` component
4. Fixes the `Button` demos to include `text` button type
5. Adds API docs to the `Button` component

My left-out issues:
1. In `Overlay` there is a property `HiddenMode` - I could not find any real used for it. I tried using it but I failed to see any result. It was added in PR #584. If you find time @mutouzdl, can you let me know the purpose of this parameter? 
2.  In `Button` there is the paramter `Search`. It looks to me that its only purpose is to add a css class `ant-input-search-button`. That css class seems to be used only `Search` component, that anyway sets it manually. It seems to me it is actually useless here, but I left it, so maybe @ElderJames could let me know what other purposes it may server (it was added by PR #82). 

Question:
How can we embed a style in docs in similar fashion as it is done in antd? For example, if you got to [antd button demo](https://ant.design/components/button/), you will notice that any button has also class 
```css
[id^=components-button-demo-] .ant-btn {
  margin-right: 8px;
  margin-bottom: 12px;
}
```
(and other) that are giving proper margins, so when the components are wrapped, they look much better.
### 📝 Changelog
| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add typical `Button` propertied to `DropdownButton`. Include demo & API docs for `Dropdown` API and `Button` API.         |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by mutouzdl** [comment]
Select component should render the overlay children on first time(but should not show). So "HiddenMode" is used to do that.

You may try to set "HiddenMode" to false in Select component's "OverlayTrigger", and check the "Simple encapsulation" demo.You should find someting wrong on it.

**Comment by anddrzejb** [comment]
@mutouzdl Thank you for the explanation. It seems this is needed only when `RenderFragment` is used to fill up the select. I will try to give it an appropriate description.

@ElderJames Something is wrong with codecov. It caused the build to fail for at least couple of last PRs.

**Comment by ElderJames** [comment]
> How can we embed a style in docs in similar fashion as it is done in antd? For example, if you got to antd button demo, you will notice that any button has also class

@anddrzejb  You can add the <style> tag in doc/*.md files

**Comment by anddrzejb** [comment]
@ElderJames This I know. It is exactly what I did for button block demo. I was hoping for a more unified solution and less verbose in the demo section, but if `<style>` section is the recommended way, then `<style>` it is.
Any advice regarding the `Search` parameter in `Button` component?

File: components/core/Component/Overlay/Overlay.razor.cs [hunk 1]
```diff
@@ -54,6 +54,14 @@ public sealed partial class Overlay : AntDomComponentBase
         [Parameter]
         public int HorizontalOffset { get; set; } = 4;
 
+        /// <summary>
```
Comment (anddrzejb) [inline components/core/Component/Overlay/Overlay.razor.cs:57]: @mutouzdl for your consideration.

Comment (mutouzdl) [reply components/core/Component/Overlay/Overlay.razor.cs:57]: I think that's a pretty clear explanation

**Comment by ElderJames** [comment]
I noticed that we have added `ant-input-search-button` to the button in the search component,  so we can delete the Search parameter.

**Review by ElderJames — APPROVED** [review]

**Pull request merged**